rio/srcrobot/commands/ExecuteCommand.py

```diff
@@ -0,0 +1,38 @@
+from commands2 import ParallelCommandGroup, RunCommand, InstantCommand
+from commands.TurnInPlace import TurnInPlace
+from subsystems.Arm import Arm
+from subsystems.Shooter import Shooter
+from subsystems.Swerve import Swerve
+
+from wpimath.geometry import Rotation2d
+
+from robotState import RobotState
+
+class ExecuteCommand(ParallelCommandGroup):
+    def __init__(self, arm : Arm, shooter : Shooter, swerve : Swerve, translationSupplier, strafeSupplier, rotationSupplier, robotCentricSupplier):
+        super().__init__()
+        self.robotState = RobotState()
+        self.arm = arm
+        self.shooter = shooter
+        self.swerve = swerve
+
+        self.arm_angle = self.robotState.m_gameState.getNextShot().m_armAngle
+        self.shooter_velocity = self.robotState.m_gameState.getNextShot().m_shooterVelocity
+        self.robot_angle = self.robotState.m_gameState.getNextShotRobotAngle()
```

Comment:
Reading these values in the constructor is very bad. We are going to always be using the values at the time the button is bound in RobotContainer. The values could be read in one of two places. In both cases, this ExecuteCommand should NOT extend ParallelCommandGroup. It should be a normal Command that is put into a parallel command group along with TurnInPlace.

The first option is to read them in initialize(). This would get the values at the time the button is pressed. Note that doing this would require cancelling the ExecuteCommand and restarting it in order to react to a next shot change while ExecuteCommand is running.

The second option is to read them in execute(). As long as the next shot is not changing, this is the same as reading in initialize(). However, if the shot is changed while ExecuteCommand is running, it will automatically adjust if the read is done in execute(). I prefer this option.

One thing we would need to carefully consider in any case is when should the ExecuteCommand terminate. We probably want it to run until we shoot or decide we want to explicitly cancel it. It appears that the shoot and autoHome buttons in RobotContainer could be doing these.

Reply:
The execute command specifically only terminated on shoot or auto home.

Also, the functions to servo the arm and setShooerVelocity are commands themselves. Is it possible to schedule a command within another command? Though I feel like that's not very clean.

Reply:
My main concern is reading the shot values during construction which happens during robotInit. If I am reading this right, the reads in initialize are not having any effect since the command group is already built and configured. Instead of using this ExecuteCommand in robotContainer, how about something like this instead (indentation is probably off as may be a couple other details)....
```
self.execute.onTrue(
            RunCommand(lambda: self.s_Arm.servoArmToTarget(self.m_robotState.m_gameState.getNextShot().m_armAngle))
            .alongWith(RunCommand(lambda: self.s_Shooter.shootVelocity(self.m_robotState.m_gameState.getNextShot().m_shooterVelocity)),
                TurnInPlace(
                    self.s_Swerve,
                    lambda: Rotation2d.fromDegrees(self.m_robotState.m_gameState.getNextShotRobotAngle()),
                    translation, strafe, rotation, robotcentric)
                        .repeatedly()))
```